vortex-expr/src/scope.rs

```diff
@@ -6,18 +6,49 @@ use vortex_array::{Array, ArrayRef};
 use vortex_dtype::DType;
 use vortex_error::{VortexResult, vortex_err};
 
-use crate::IDENTITY_IDENTIFIER;
-
-pub type Identifier = Arc<str>;
 type ExprScope<T> = HashMap<Identifier, T>;
 
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Identifier {
+    Identity,
+    Other(Arc<str>),
```

Comment:
There's also `IdentityOrField` in the pruning system which is maybe isomorphic to this. Should this just be a FieldPath? @joseph-isaacs are you working on something like that?

Reply:
https://github.com/vortex-data/vortex/pull/3476